hack/go-build.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+
+set -eu
+
+REPO=github.com/openshift/machine-api-operator
+WHAT=${1:-machine-api-operator}
+GOFLAGS=${GOFLAGS:-}
+GLDFLAGS=${GLDFLAGS:-}
+
+eval $(go env | grep -e "GOHOSTOS" -e "GOHOSTARCH")
+
+GOOS=${GOOS:-${GOHOSTOS}}
+GOARCH=${GOACH:-${GOHOSTARCH}}
+
+# Go to the root of the repo
+cd "$(git rev-parse --show-cdup)"
+
+if [ -z ${VERSION_OVERRIDE+a} ]; then
+	echo "Using version from git..."
+	VERSION_OVERRIDE=$(git describe --abbrev=8 --dirty --always)
+fi
+
+GLDFLAGS+="-extldflags '-static' -X ${REPO}/pkg/version.Raw=${VERSION_OVERRIDE}"
+
+eval $(go env)
+
+mkdir -p bin
```

Comment:
`go build` will create the `-o` directories if they don't exist so you can drop this.